fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2444,13 +2447,32 @@ private Bundle createSearchBundle(List<Resource> resources, FHIRSearchContext se
                                             .id(UUID.randomUUID().toString())
                                             .total(totalCount);
 
+        // Calculate how many resources are 'match' mode
+        int pageSize = searchContext.getPageSize();
+        int offset = (searchContext.getPageNumber() - 1) * pageSize;
+        int matchResourceCount = pageSize;
+        if (totalCount.getValue() < offset + pageSize) {
+            matchResourceCount = totalCount.getValue() - offset;
+        }
+
+        // Check if too many included resources
+        if (resources.size() > matchResourceCount + SearchConstants.MAX_PAGE_SIZE) {
+            String msg = "Number of returned 'include' resources exceeds allowable limit of " + SearchConstants.MAX_PAGE_SIZE;
```

Comment:
seems like this could actually be a final and possibly even final static at the class level.

Reply:
Moved to final static at class level.